IQSS/dataverse-frontend
# Featured Items - Collection Page

## What this PR does / why we need it:
Add featured items section in the Collection Page.
Allows users that can edit a collection to create "featured items" that will initially support content and an optional image.
These "featured items" will be shown between the Collection page header and the items list.

Storybook links:
- https://646f68aa9beb01b35c599acd-nvjhuorxtc.chromatic.com/?path=/story/pages-collection--with-featured-items
- https://646f68aa9beb01b35c599acd-nvjhuorxtc.chromatic.com/?path=/story/pages-edit-collection-featured-items--default

## Which issue(s) this PR closes:

- Closes #568

## Suggestions on how to test this:
**Step 1: Run the Development Environment**
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm i && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

**Step 2: Test the feature**
1. Login and create a new collection or even in the root collection click the Edit Collection > Featured Items button.
2. Create featured items, with different contents, with and without image and submit the form, change the order of the featured items by dragging them with the drag handle.
3. Check that the featured items shown in the collection page are correct.
4. Go to edit them again, replace images from a featured item, also you can test removing an image from a featured item and also remove a featured item by clicking the minus button. *Note: you can remove the first item, to be removed needs to be at least in the 2nd position.
5. Check that the featured items shown in the collection page are correct.
6. Go to edit them again but now try the delete all featured items action, when clicking it, a confirmation dialog will appear.
7. Check in the collection page that no featured items are shown.
8. About field validations: the content field cannot be empty and must be 15,000 characters or less, but note that if you enter "Hello world" in the field, this rich text editor will transform that into `<p class="rte-paragraph">Hello world</p>` so those html tags characters also count for the validation.
9. You could also validate that if a collection is unpublished and has featured items, only users that can view unpublished collections can see the featured items.

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
Yes.
![Screen Shot 2025-01-02 at 08 49 32](https://github.com/user-attachments/assets/5b1862ae-fcf0-4aa1-afbf-205151f8ce1b)
![Screen Shot 2025-01-02 at 08 55 29](https://github.com/user-attachments/assets/6939a24a-8fa2-497e-bffa-244454874a98)
![Screen Shot 2025-01-02 at 08 55 40](https://github.com/user-attachments/assets/4a3ea5d5-78fb-4386-af96-219b827d2863)


## Is there a release notes update needed for this change?:

## Additional documentation:
